src/java.base/share/classes/sun/security/util/DisabledAlgorithmConstraints.java

```diff
@@ -267,17 +264,17 @@ public final void permits(String algorithm, ConstraintsParameters cp,
         algorithmConstraints.permits(algorithm, cp, checkKey);
     }
 
-    private static List<String> getNamedCurveFromKey(Key key) {
-        if (key instanceof ECKey) {
-            NamedCurve nc = CurveDB.lookup(((ECKey)key).getParams());
-            return (nc == null ? List.of()
-                               : Arrays.asList(nc.getNameAndAliases()));
-        } else if (key instanceof XECKey) {
-            return List.of(
-                ((NamedParameterSpec)((XECKey)key).getParams()).getName());
-        } else {
-            return List.of();
-        }
+    private static List<String> getNamedParametersFromKey(Key key) {
+        return switch (key) {
+            case ECKey ecKey -> {
+                NamedCurve nc = CurveDB.lookup(ecKey.getParams());
+                if (nc == null) {
+                    yield List.of();
+                }
+                yield List.of(nc.getNameAndAliases());
+            }
+            default -> List.of(key.getAlgorithm(), KeyUtil.getAlgorithm(key));
```

Comment:
What if these 2 are the same string?

Reply:
Are you concerned about duplicate checks?

Reply:
Yes.